src/unpage/utils.py

```diff
@@ -256,25 +255,39 @@ async def checkbox(
 def import_submodules(
     package: str | ModuleType,
     recursive: bool = True,
-) -> dict[str, ModuleType]:
+) -> None:
     """Import all submodules of a module, recursively, including subpackages
 
+    Uses filesystem scanning to avoid namespace pollution from installed packages
+    with similar names.
+
     :param package: package (name or actual module)
     :type package: str | module
     :rtype: dict[str, types.ModuleType]
     """
     if isinstance(package, str):
         package = importlib.import_module(package)
-    results: dict[str, ModuleType] = {}
-    for _, name, is_pkg in pkgutil.walk_packages(package.__path__):
-        full_name = f"{package.__name__}.{name}"
-        try:
-            results[full_name] = importlib.import_module(full_name)
-        except ModuleNotFoundError:
+
+    # Get the actual filesystem modules only to avoid namespace pollution
+    package_dir = Path(package.__path__[0])  # Get the first path
+
+    for item_path in package_dir.iterdir():
+        # Skip hidden files and private modules.
+        if item_path.name[0] in ("_", "."):
             continue
-        if recursive and is_pkg:
-            results.update(import_submodules(full_name))
-    return results
+
+        name = item_path.stem
+        is_pkg = item_path.is_dir() and (item_path / "__init__.py").exists()
+
+        # Skip non-Python files and folders.
+        if not is_pkg and item_path.suffix != ".py":
+            continue
+
+        full_name = f"{package.__name__}.{name}"
+        importlib.import_module(full_name)
+
+        if is_pkg and recursive:
+            import_submodules(full_name)
```

Comment:
I discovered a bug with the previous `pkgutils.walk_packages` approach, wherein submodules that share names with other installed packages (e.g., `unpage.plugins.pagerduty` shares a base name with the `pagerduty` package) it would try to import the top-level members of the installed package. Harmless, but annoying -- except that the `ModuleNotFound` error that was being caught also silently swallowed import errors while I was refactoring.

Anyway, this still works as it used to, but won't swallow import problems anymore.